lib/jwt/encode.rb

```diff
@@ -1,51 +1,76 @@
 # frozen_string_literal: true
 
-require 'json'
-
 # JWT::Encode module
 module JWT
   # Encoding logic for JWT
   class Encode
-    attr_reader :payload, :key, :algorithm, :header_fields, :segments
+    ALG_NONE = 'none'.freeze
```

Comment:
Wouldn't `nil` be a better representation of none?

Reply:
This is something related to creating a JWT without a signature. none is the algorithm the [JWA spec](https://tools.ietf.org/html/rfc7518#section-3.6) defines as "Unsecured JWS"